### PR TITLE
Improve QR code formatting documentation

### PR DESCRIPTION
**Pull request by Kixunil**
It wasn't obvious that displaying address with alternate formatting upper cases bech32 addresses.

This change adds information about this and also a note about the compatibility of various wallets.

BTW why do we prefix upper case `BITCOIN` only in case of bech32? Proper BIP21 should support it for all addresses. (And yest, it still does save space.)

### CODE REVIEW

**Comment by apoelstra** [comment]
@Kixunil other addresses are case sensitive. You can't uppercase them.

**Review by apoelstra — APPROVED** [review]
ACK a6188a90c0bade6a0df16feaafe81d24328ae7da

File: bitcoin/src/address.rs [hunk 1]
```diff
@@ -937,6 +938,26 @@ impl Address {
...
+    /// # let address = ADDRESS.parse::<bitcoin::Address<_>>().unwrap().assume_checked();
+    /// # let mut writer = String::new();
+    /// # // magic trick to make error handling look better
+    /// # (|| -> Result<(), core::fmt::Error> {
+    ///
+    /// write!(writer, "{:#}", address)?;
+    ///
+    /// # Ok(())
+    /// # })().unwrap();
+    /// # assert_eq!(writer, ADDRESS);
```
Comment (tcharding) [inline bitcoin/src/address.rs:959]: Did you mean to hide the assertion bro, usually we leave that as part of the docs?

**Review by tcharding — APPROVED** [review]
ACK a6188a90c0bade6a0df16feaafe81d24328ae7da

File: bitcoin/src/address.rs (hunk 1 above)
Comment (Kixunil) [reply bitcoin/src/address.rs:959]: Yes, I did to not confuse readers.

**Pull request merged**